### PR TITLE
Add sanitizer option

### PR DESCRIPTION
**Pull request by Meakk**

### CODE REVIEW

File: cmake/testing.cmake [hunk 1]
```diff
@@ -40,13 +40,17 @@ function(f3d_test)
...
-    if(NOT F3D_ENABLE_LONG_TIMEOUT_TESTS)
-      set_tests_properties(${F3D_TEST_NAME} PROPERTIES DISABLED ON)
-    endif()
+  if(F3D_SANITIZER)
+    set_tests_properties(${F3D_TEST_NAME} PROPERTIES TIMEOUT 3600)
   else()
-    set_tests_properties(${F3D_TEST_NAME} PROPERTIES TIMEOUT 30)
+    if(F3D_TEST_LONG_TIMEOUT OR F3D_TEST_INTERACTION)
+      set_tests_properties(${F3D_TEST_NAME} PROPERTIES TIMEOUT 120)
+      if(NOT F3D_ENABLE_LONG_TIMEOUT_TESTS)
```
Comment (mwestphal) [inline cmake/testing.cmake:48]: enabling/disabling long timeout and enabling/disabling sanitizer should both be possible.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -266,3 +266,94 @@ jobs:
...
+#----------------------------------------------------------------------------
+# Sanitizer: Build and test on linux with last VTK with sanitizer options
+#----------------------------------------------------------------------------
+  sanitizer:
+    if: github.event.pull_request.draft == false
+
+    strategy:
+      fail-fast: false
+      matrix:
+        # "memory" is not supported by the GCC version in our Docker image yet
```
Comment (mwestphal) [inline .github/workflows/ci.yml:279]: we can try to update it if you want

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -266,3 +266,94 @@ jobs:
...
+    - name: Install Alembic dependency
+      uses: ./source/.github/actions/alembic-install-dep
+
+    - name: Install VTK dependency
+      uses: ./source/.github/actions/vtk-install-dep
+      with:
+        vtk_sha_file: ./source/.github/actions/vtk-install-dep/vtk_commit_sha
+
+    # pwd seems necessary for OpenCASCADE and Assimp for some reasons
+    # coverage build is done in source as it seems to be required for codecov
```
Comment (mwestphal) [inline .github/workflows/ci.yml:331]: this is not coverage, and is in source build needed for lsan ?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -266,3 +266,94 @@ jobs:
...
+    - name: Build
+      working-directory: ${{github.workspace}}/source
+      run: cmake --build . --parallel 2 --config Release
+
+    - name: Run Xvfb
+      run: Xvfb $DISPLAY -screen 0 1280x1024x24 &
+
+    - name: Test
+      working-directory: ${{github.workspace}}/source
+      run: ctest -C Release -j 2 --output-on-failure || ctest -C Release -j 1 --rerun-failed
```
Comment (mwestphal) [inline .github/workflows/ci.yml:359]: so the test will fail is a lsan error is detected ?

**Comment by mwestphal** [comment]
please try introducing a memory error and show that the CI fail in that case.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (Meakk) [reply .github/workflows/ci.yml:279]: actually, it's supported by clang only

File: .github/workflows/ci.yml (hunk 3 above)
Comment (Meakk) [reply .github/workflows/ci.yml]: fixed

**Comment by Meakk** [comment]
> please try introducing a memory error and show that the CI fail in that case.

https://github.com/f3d-app/f3d/runs/4826250319?check_suite_focus=true#step:16:363

Failing as expected considering the leak I introduced

File: .github/workflows/ci.yml (hunk 2 above)
Comment (mwestphal) [reply .github/workflows/ci.yml:279]: should we use clang to build then ?

**Comment by mwestphal** [comment]
> Failing as expected considering the leak I introduced

Nice !

**Review by mwestphal — APPROVED** [review]

**Pull request merged**